src/navigation/pathnavigator.cpp

```diff
@@ -0,0 +1,439 @@
+}
+
+void PathNavigator::updateCamera(double deltaTime) {
+    ghoul_assert(camera() != nullptr, "Camera must not be nullptr");
+
+    if (!hasCurrentPath()) {
+        return;
+    }
+
+    if (!_isPlaying) {
+        return;
+    }
+
+    // Prevent long delta times due to e.g. computations from other actions to cause
+    // really big jumps in the motion along the path
+    // OBS! Causes problems if the general FPS is lower than 10, but then the user should
+    // probably not use the camera paths anyways
+    if (deltaTime > 0.1) {
+        deltaTime = 0.01;
+    }
+
+    // If for some reason the time is no longer paused, pause it again
+    // TODO: Before we get here, one time tick happens. Should move this check to engine
+    if (!global::timeManager->isPaused()) {
+        global::timeManager->setPause(true);
+        LINFO("Cannot start simulation time during camera motion");
+    }
+
+    CameraPose newPose = _currentPath->traversePath(deltaTime, _speedScale);
+    const std::string newAnchor = _currentPath->currentAnchor();
+
+    // Set anchor node in orbitalNavigator, to render visible nodes and add activate
+    // navigation when we reach the end.
+    std::string currentAnchor = anchor()->identifier();
+    if (currentAnchor != newAnchor) {
+        global::navigationHandler->orbitalNavigator().setFocusNode(newAnchor, false);
+    }
+
+    if (!_includeRoll) {
+        removeRollRotation(newPose, deltaTime);
+    }
+
+    camera()->setPositionVec3(newPose.position);
+    camera()->setRotation(newPose.rotation);
+
+    if (_currentPath->hasReachedEnd()) {
+        LINFO("Reached end of path");
+        _isPlaying = false;
+
+        if (_applyIdleBehaviorOnFinish) {
+            constexpr const char* ApplyIdleBehaviorScript =
+                "local f = 'NavigationHandler.OrbitalNavigator.IdleBehavior.ApplyIdleBehavior';"
+                "openspace.setPropertyValueSingle(f, true);";
+
+            global::scriptEngine->queueScript(
+                ApplyIdleBehaviorScript,
+                openspace::scripting::ScriptEngine::RemoteScripting::Yes
+            );
+        }
+        return;
+    }
+}
+
+void PathNavigator::createPath(const ghoul::Dictionary& dictionary) {
+    // TODO: Improve how curve types are handled
+    const int curveType = _defaultCurveOption;
+
+    // Ignore paths that are created during session recording, as the camera
+    // position should have been recorded
+    if (global::sessionRecording->isPlayingBack()) {
+        return;
+    }
+
+    clearPath();
+    try {
+        _currentPath = std::make_unique<Path>(
+            createPathFromDictionary(dictionary, Path::CurveType(curveType))
+        );
+    }
+    catch (const ghoul::RuntimeError& e) {
+        LERROR(fmt::format("Could not create path. Reason: ", e.message));
+        return;
+    }
+
+    LINFO("Successfully generated camera path");
+}
+
+void PathNavigator::clearPath() {
+    LINFO("Clearing path...");
```

Comment:
I tried to get rid of the ellipses in messages as it looks a bit weird in the log files